ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -326,7 +328,23 @@ calcParserModules ghcFlavor = do
         [ if ghcFlavor `elem` [ GhcMaster, Ghc8101 ] then "GHC.Hs.Dump" else "HsDumpAst"]
   return $ nubSort (modules ++ extraModules)
 
--- | Selectively disable optimizations in some particular files so as
+-- Rename 'HsVersions.h' to 'GhclibHsVersions.h' then replace
+-- occurences of that string in all .hs and .y files reachable from
+-- the 'compiler' directory. This enables use of ghc-lib-parser with
+-- ghcjs (issue https://github.com/digital-asset/ghc-lib/issues/204).
+applyPatchHsVersions :: GhcFlavor -> IO ()
+applyPatchHsVersions _ = do
+  renameFile "compiler/HsVersions.h" "compiler/GhclibHsVersions.h"
+  files <- find always (extension ==? ".hs" ||? extension ==? ".y") "compiler"
```

Comment:
`listFilesRecursive` + `filter` seems simpler and avoids the extra dependency.

Reply:
Ah yes. Good catch.